packages/@react-stately/list/src/ListCollection.ts

```diff
@@ -18,17 +18,27 @@ export class ListCollection<T> implements Collection<Node<T>> {
   private iterable: Iterable<Node<T>>;
   private firstKey: Key;
   private lastKey: Key;
+  sections: Node<T>[];
+  rows: Node<T>[];
```

Comment:
TODO: try removing these so migrating to RAC is easier. Have the warning for mixed non-section rows and sections happen in here instead?